returns/contrib/hypothesis/laws.py

```diff
@@ -10,22 +10,48 @@
 from hypothesis.strategies._internal import types  # noqa: PLC2701
 
 from returns.contrib.hypothesis.containers import strategy_from_container
+from returns.contrib.hypothesis.type_resolver import (
+    StrategyFactory,
+    strategies_for_types,
+)
 from returns.primitives.laws import LAWS_ATTRIBUTE, Law, Lawful
 
+Example_co = TypeVar('Example_co', covariant=True)
+
 
 @final
-class _Settings(NamedTuple):
+class Settings(NamedTuple):
```

Comment:
Let's also document `Settings` in `hypothesis_plugins.rst` since it is now public.

Reply:
I'd made it public just so that I could use it in tests :P. I'm making it private again, because `check_all_laws` accepts direct arguments, not a settings object. So, I don't think it makes much sense to document the class in public and unnecessarily make the class harder to change in the future. Lmk if you feel strongly.